pkg/server/controllers.go

```diff
@@ -374,19 +362,19 @@ func (s *Server) installWorkspaceScheduler(ctx context.Context, config *rest.Con
 
 	workspaceController, err := clusterworkspace.NewController(
 		kcpClusterClient,
-		s.kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces(),
-		s.rootKcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaceShards(),
-		s.kcpSharedInformerFactory.Apis().V1alpha1().APIBindings(),
+		s.KcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces(),
+		s.KcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaceShards(),
```

Comment:
am I right that `ClusterWorkspaceShards` are only in the root `ws`?

Reply:
yes. But for that we should use informer scoping if we want that.